pkg/snmp/snmp_test.go

```diff
@@ -160,6 +160,117 @@ snmp_listener:
 	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
 }
 
+func TestNewNetworkDevicesListenerConfig(t *testing.T) {
+	config.Datadog.SetConfigType("yaml")
+
+	// default collect_device_metadata should be true
+	err := config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err := NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, true, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+
+	// collect_device_metadata: false
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    collect_device_metadata: false
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err = NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, false, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+
+	// collect_device_metadata: true
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    collect_device_metadata: true
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err = NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, true, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+}
+
+func TestBothListenersConfig(t *testing.T) {
```

Comment:
I think this is the most important test in this PR. We should likely extend it to test for more use cases:
1. What if snmp_listener is defined (but incorrect) while the other is defined and correct?
2. What if snmp_listener is correct but the other is incorrect.
3. Asserting that this is not extended but instead only the new one takes precedence.

And there might be a few other cases to think of

Reply:
I extended this test in [this commit](https://github.com/DataDog/datadog-agent/pull/24644/commits/e16b349fef49ccfe769bb2d561a140e1a4489d4f)